### PR TITLE
Support for optional arguments

### PR DESCRIPTION
**Pull request by pitchart**
I'm sending a PR so we can talk about the code if needed.

I'm not convinced by the tests function names

### CODE REVIEW

File: src/Reference.php [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+     */
+    public function nullIfMissing() : self
+    {
+        return new static((string) $this, ContainerInterface::NULL_ON_INVALID_REFERENCE);
+    }
+
+    /**
+     * Same as nullIfMissing() but the method call is removed in case of setter injection
+     *
+     * @see https://symfony.com/doc/current/service_container/optional_dependencies.html#ignoring-missing-dependencies
```
Comment (rejinka) [inline src/Reference.php]: see above

File: src/Reference.php [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+use Symfony\Component\DependencyInjection\ContainerInterface;
+use Symfony\Component\DependencyInjection\Reference as SymfonyReference;
+
+class Reference extends SymfonyReference
+{
+    /**
+     * Reference to another service or null if it does not exist
+     *
+     * @see https://symfony.com/doc/current/service_container/optional_dependencies.html#setting-missing-dependencies-to-null
```
Comment (rejinka) [inline src/Reference.php]: "@link" instead of "@see"

**Comment by mnapoli** [comment]
That looks good to me (and anyway it's a rather restricted use case), @rejinka good to merge for you too?

**Comment by mnapoli** [comment]
(a rebase is going to be needed here too)

**Comment by pitchart** [comment]
Merge done

**Comment by mnapoli** [comment]
Thanks!

**Pull request merged**